internal/cephfs/controllerserver.go

```diff
@@ -1226,32 +1230,69 @@ func (cs *ControllerServer) ControllerUnpublishVolume(
 		}
 	}
 
-	credentials, err := util.NewAdminCredentials(secrets)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to create credentials from secrets: %v", err)
-	}
-	defer credentials.DeleteCredentials()
-
 	volOptions, _, err := store.NewVolumeOptionsFromVolID(ctx, volumeId, nil, secrets, cs.ClusterName, cs.SetMetadata)
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "failed to generate volume from volume ID %s: %v", volumeId, err)
 	}
 	defer volOptions.Destroy()
 
-	err = cs.fenceNode(ctx, req.GetNodeId(), volOptions, credentials)
+	err = cs.fenceNode(ctx, volumeId, req.GetNodeId(), secrets, volOptions)
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
 	return &csi.ControllerUnpublishVolumeResponse{}, nil
 }
 
+// getSubvolumeMetadataHandler returns the appropriate metadata handler based on whether
+// the volume is a backing snapshot or a regular subvolume.
+func (cs *ControllerServer) getSubvolumeMetadataHandler(
+	ctx context.Context,
+	volOptions *store.VolumeOptions,
+	secrets map[string]string,
+) (*subvolumeMetadataHandler, error) {
+	conn := volOptions.GetConnection()
+	handler := &subvolumeMetadataHandler{}
```

Comment:
create and return handler within if condition or outside.

only 
```
{
..
if backsnapshot {

return 
}

return 
}
```

is sufficient ?